### PR TITLE
OEUI-56: fix careSetting undefined error

### PR DESCRIPTION
**Pull request by fuchodeveloper**
[OEUI-56: Integrate care settings into tab functionality](https://issues.openmrs.org/projects/OEUI/issues/OEUI-56)

SUMMARY
Fix `careSetting` undefined error

### CODE REVIEW

File: app/js/components/orderEntry/OrderEntryPage.jsx [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
       </div>
     );
   }
 }
 
-export default OrderEntryPage;
+const mapStateToProps = ({ careSettingReducer }) => ({
+  inpatientCareSetting: careSettingReducer.patientCareSetting,
+  outpatientCareSetting: careSettingReducer.patientCareSetting,
+});
```
Comment (zeze111) [inline app/js/components/orderEntry/OrderEntryPage.jsx]: is there a reason why you separated the patientCareSettings, if they're both receiving the same data from the reducer?

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Have modified my implementation to allow for fetching `inpatientCareSetting` and `outpatientCareSetting` explicitly. 
Thanks for pointing that out.

File: app/js/components/orderEntry/OrderEntryPage.jsx [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+  componentDidMount() {
+    this.props.fetchPatientCareSetting();
+  }
 
-class OrderEntryPage extends React.Component {
   render() {
+    if (!(this.props.outpatientCareSetting || this.props.inpatientCareSetting)) {
+      return (
+        <div>
+          loading...
```
Comment (flacode) [inline app/js/components/orderEntry/OrderEntryPage.jsx]: @fuchodeveloper, I like the way you are using one endpoint for both careSettings, though the `<div>loading</div>` would look better visually say using a spinner than verbal.

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Will consult the team concerning the idea of using a reusable spinner in scenarios such as this and update this PR based on the team's response.

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Updated now to make use of the default OpenMRS spinner image based on the team's decision.

**Comment by fuchodeveloper** [comment]
@dkayiwa 
Yes. The PR aims to solve the error while also optimizing the code better.

**Comment by dkayiwa** [comment]
I would use separate pull requests for each.

**Comment by fuchodeveloper** [comment]
@dkayiwa 
If I understand correctly, you recommend I split the fix and optimization into two pull requests.

**Comment by fuchodeveloper** [comment]
Ok
I would make the necessary changes

**Comment by fuchodeveloper** [comment]
Closing this PR based on feedback from @dkayiwa. The PR containing the implementations from the feedback can be found [here](https://github.com/openmrs/openmrs-owa-orderentry/pull/21)

**Pull request closed without merging**